fem/coefficient.hpp

```diff
@@ -723,6 +729,14 @@ class MatrixCoefficient
    virtual void Eval(DenseMatrix &K, ElementTransformation &T,
                      const IntegrationPoint &ip) = 0;
 
+   /** @brief Evaluate the upper triangular entries of the matrix coefficient
+       in the symmetric case, similarly to Eval. Matrix entry (i,j) is stored
+       in K[j - i + os_i] for 0 <= i <= j < width, os_0 = 0,
+       os_{i+1} = os_i + width - i. */
```

Comment:
This documentation is good but perhaps something more visual could be added.  Something like:
K = {M(0,0), M(0,1), ..., M(0,w-1), M(1,1), M(1,2), ..., M(1,w-1),..., M(w-1,w-1)};